framework/packages/abstract-interface/src/account/mod.rs

```diff
@@ -313,3 +432,14 @@ impl<Chain: CwEnv> std::fmt::Display for AbstractAccount<Chain> {
         )
     }
 }
+
+fn are_versions_compatible(v1: &Version, v2: &Version) -> bool {
+    if v1.major == 0 && v2.major == 0 {
+        // For major version 0, compatibility can be more strict; typically, versions are considered
+        // compatible if they have the same minor version
+        v1.minor == v2.minor
+    } else {
+        // For major version >= 1, versions are compatible if they have the same major version
+        v1.major == v2.major
+    }
+}
```

Comment:
It's different logic from the one we have in `assert_contract_upgrade`, should we update logic there as well? Wanna add source for the comment? https://semver.org/#spec-item-4

Reply:
What do you think of using the semver crate implementation instead of trying to implement it ourselves (just so we're sure). 

I.e. parse the v1 version into a `VersionReq` type and then assert v2 against it: https://docs.rs/semver/latest/semver/struct.VersionReq.html#method.parse